### PR TITLE
Native support for CoreOS.

### LINKED ISSUE #99: Suggestion: get rid of the CoreOS / "buddy"
While flexible, the buddy / `hosts` file things complicates the deployment.
I ended up moving Fleet logic into Heapster and getting rid of noticeable chunk of the code. As there are no currently other types of the buddies (Mesos?), I suggest doing something like the following:
https://github.com/arkadijs/heapster/commit/7c4fc6ca628a2e9548fd26241993f5d30dba301e

1. If there is Kubernetes - use listMinions
2. Otherwise try Fleet.

The code is little bit messy - coupling KubeSource to ExternalSource, but my baseline is CoreOS with Deis, with optional Kubernetes, thus such practical solution.

### PR DESCRIPTION
**Pull request by vishh**
Fixes #99

### CODE REVIEW

**Comment by vmarmol** [comment]
```
$ godep go test -a -test.short ./...
# github.com/GoogleCloudPlatform/heapster/sources/nodes
sources/nodes/external_test.go:53: too few values in struct initializer
?   	github.com/GoogleCloudPlatform/heapster	[no test files]
?   	github.com/GoogleCloudPlatform/heapster/clusters/coreos	[no test files]
ok  	github.com/GoogleCloudPlatform/heapster/integration	0.016s
?   	github.com/GoogleCloudPlatform/heapster/sinks	[no test files]
?   	github.com/GoogleCloudPlatform/heapster/sinks/gcm	[no test files]
ok  	github.com/GoogleCloudPlatform/heapster/sources	0.018s
?   	github.com/GoogleCloudPlatform/heapster/sources/api	[no test files]
ok  	github.com/GoogleCloudPlatform/heapster/sources/datasource	0.019s
FAIL	github.com/GoogleCloudPlatform/heapster/sources/nodes [build failed]
?   	github.com/GoogleCloudPlatform/heapster/validate	[no test files]
?   	github.com/GoogleCloudPlatform/heapster/version	[no test files]
godep: go exit status 2
The command "godep go test -a -test.short ./..." exited with 1.
```

**Comment by vishh** [comment]
@vmarmol @rjnagal: I want to include this in the v0.7 release. I tested manually and it works. I have updated the readme and release notes.

File: sources/nodes/coreos.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	if self.nodes != nil {
+		output = fmt.Sprintf("%s\nCadvisor Nodes: %v", output, self.nodes.Items)
+	}
+	return output
+}
+
+func getFleetRegistryClient(fleetEndpoints []string) (fleetClient.API, error) {
+	var dial func(string, string) (net.Conn, error)
+
+	tlsConfig, err := fleetPkg.ReadTLSConfigFiles("", "", "")
```
Comment (vmarmol) [inline sources/nodes/coreos.go]: This won't let us do HTTPS no? What is the alternative to this? Do you actually have to give it the file paths?

**Comment by vmarmol** [comment]
LGTM, just a question we can continue in another PR

File: sources/nodes/coreos.go (hunk 1 above)
Comment (vishh) [inline sources/nodes/coreos.go]: That is true. If users set up https, then we will have to pass auth info. I don't want to do it unless some one needs it.

Comment (vmarmol) [inline sources/nodes/coreos.go]: I *think* we can just get this from the filesystem.

Comment (vishh) [inline sources/nodes/coreos.go]: Depends on if we have access to the 'filesystem'.

Comment (vmarmol) [inline sources/nodes/coreos.go]: We do from the container :)

I have to send a PR to add a hostVolume for Heapster so we can use HTTPS. We need it for GCM and I forgot to add it :P

Comment (vishh) [inline sources/nodes/coreos.go]: Great!

**Pull request merged**